presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -154,9 +156,19 @@ protected Node visitShowStats(ShowStats node, Void context)
                 throw rewriteAccessDeniedException(e);
             }
 
+            for (ColumnHandle column : columnHandles.values()) {
+                ColumnMetadata columnMetadata = metadata.getColumnMetadata(session, tableHandle, column);
+                if (!accessControl.getColumnMasks(session.toSecurityContext(), tableName, columnMetadata.getName(), columnMetadata.getType()).isEmpty()) {
+                    throw new PrestoException(NOT_SUPPORTED, "SHOW STATS for table with column masking is not supported: " + columnMetadata.getName());
+                }
+            }
+            if (!accessControl.getRowFilters(session.toSecurityContext(), tableName).isEmpty()) {
+                throw new PrestoException(NOT_SUPPORTED, "SHOW STATS is not supported for a table with row filtering");
```

Comment:
maybe add
```
// SHOW STATS could reveal min, max value which would be otherwise filtered out; or could reveal statistical properties of the data, like row count
```